### PR TITLE
Added suspend/resume to the device driver

### PR DESCRIPTION
**Pull request by ksaihtam**

### CODE REVIEW

**Comment by Tragetaschen** [comment]
Please use tabs instead of spaces

File: drivers/target-systemelektronik/fpga.c [hunk 1]
```diff
@@ -434,6 +434,34 @@ static void fpga_driver_remove(struct pci_dev *dev)
 	fpga_teardown_irq(dev);
 }
 
+#ifdef CONFIG_PM
+static int fpga_driver_suspend (struct pci_dev *pdev, pm_message_t state)
+{
+        struct net_device *dev = pci_get_drvdata (pdev);
```
Comment (Tragetaschen) [inline drivers/target-systemelektronik/fpga.c:440]: Why?

File: drivers/target-systemelektronik/fpga.c [hunk 2]
```diff
@@ -434,6 +434,34 @@ static void fpga_driver_remove(struct pci_dev *dev)
...
+        pci_save_state(pdev);
+        pci_disable_device(pdev);
+        pci_set_power_state(pdev, pci_choose_state(pdev, state));
+
+        return 0;
+}
+
+static int fpga_driver_resume (struct pci_dev *pdev)
+{
+        struct net_device *dev = pci_get_drvdata (pdev);
```
Comment (Tragetaschen) [inline drivers/target-systemelektronik/fpga.c:451]: Why?

File: drivers/target-systemelektronik/fpga.c [hunk 3]
```diff
@@ -434,6 +434,34 @@ static void fpga_driver_remove(struct pci_dev *dev)
...
+        int rc;
+
+        pci_set_power_state(pdev, PCI_D0);
+        pci_restore_state(pdev);
+
+        rc = pci_enable_device(pdev);
```
Comment (Tragetaschen) [inline drivers/target-systemelektronik/fpga.c:457]: Just return the return code without `rc`?

**Comment by Tragetaschen** [comment]
Also, please rebase this against the `trizeps7` branch (and create a new PR…) as this is the one used for the F500. I know, naming is a mess and on my list of TODOs.

File: drivers/target-systemelektronik/fpga.c (hunk 1 above)
Comment (ksaihtam) [inline drivers/target-systemelektronik/fpga.c:440]: copy/paste from network driver

File: drivers/target-systemelektronik/fpga.c (hunk 2 above)
Comment (ksaihtam) [inline drivers/target-systemelektronik/fpga.c:451]: copy/paste from network driver

File: drivers/target-systemelektronik/fpga.c (hunk 3 above)
Comment (ksaihtam) [inline drivers/target-systemelektronik/fpga.c:457]: copy/paste from network driver

**Pull request closed without merging**